### PR TITLE
Adds configuration options for sendInterval and sendBatchSize

### PR DESCRIPTION
**Pull request by timbotnik**
Based on our needs and the issues #35 and #73 (perhaps others), this PR adds the ability to set a shorter than 15s send interval, as well as a configurable batch size to send more than one notification per interval.  One could argue for a more elaborate refactoring of this whole interval concept, but this should suffice for now.

### CODE REVIEW

**Comment by oskarszoon** [comment]
The "slowness" of the notifications was driving me crazy until I found the sendInterval and this branch. Thanks @timbotnik!

I'm running directly off your branch, and the app and notifications are working fine (with a 500ms interval and 50 batch size). But I would appreciate being able to move back to the regular package of course.

File: lib/server/push.api.js [hunk 1]
```diff
@@ -443,10 +443,16 @@ Push.Configure = function(options) {
...
-            isSendingNotification = true;
+        if (isSendingNotification) {
+            return;
+        }
+        // Set send fence
+        isSendingNotification = true;
 
+        var countSent = 0;
+        var batchSize = options.sendBatchSize || 1;
+        while (Push.notifications.find().count() > 0 && countSent++ < batchSize) {
```
Comment (raix) [inline lib/server/push.api.js]: we can't rely on notifications being removed - theres an option for keeping sent notes. We could how ever do something like:
```js
// Find notifications that is not being or already sent
var pendingNotifications = Push.notifications.find({ $and: [
    // Message is not sent
    { sent : { $ne: true } },
    // And not being sent by other instances
    { sending: {$ne: true} },
  ]}, {
    // Sort by created date
    sort: { createdAt: 1 },
    limit: batchSize
  });

pendingNotifications.forEach(function(notification) {
  /* reserve and send... */
});
```

--

Mental note: We should ensure index on `createdAt` etc.

Comment (timbotnik) [inline lib/server/push.api.js]: @raix that definitely makes sense, I was thinking a single query per batch would be a good idea as well.  I’ll let you know when I have some time to work on it, unless someone else does it first :-)

**Comment by timbotnik** [comment]
@raix - updated this PR as discussed.

**Comment by raix** [comment]
Thanks @timbotnik

**Comment by raix** [comment]
please help test 2.6.12-rc1

**Pull request merged**